### PR TITLE
Shown public_to_mm profiles when requesting user is enrolled in one of the programs where profile user is enrolled

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
https://github.com/mitodl/micromasters/issues/2947#issuecomment-296650753

#### What's this PR do?
- Now non staff users can view  public_to_mm profiles, when they are enrolled in one of the programs where profile user is enrolled.
- Staff can already view profiles.

#### How should this be manually tested?
- Create a learner and try to view profile of other learner in same program.
- Create a learner and try to view profile of other learner in other program.

@pdpinch

### CODE REVIEW

File: profiles/permissions_test.py [hunk 1]
```diff
@@ -204,20 +204,51 @@ def test_can_view_if_verified_mm_user(self):
...
+                user=self.profile_user,
+            )
 
         with mute_signals(post_save):
-            verified_user = UserFactory.create()
-            username = "{}_edx".format(verified_user.username)
-            verified_user.social_auth.create(
+            user = UserFactory.create()
+            username = "{}_edx".format(user.username)
+            user.social_auth.create(
```
Comment (rhysyngsun) [inline profiles/permissions_test.py]: I'd suggest writing a factory for the social auth creation here. I see very similar code repeated else where and think it's worth writing that now so we can eliminate a bunch of that boilerplate. That factory would probably live best in the `micromasters` module.

**Comment by rhysyngsun** [comment]
Functionality looks good.

File: profiles/permissions_test.py (hunk 1 above)
Comment (amir-qayyum-khan) [inline profiles/permissions_test.py]: This is out of scope, i think we should handle this in a separate task.
please create an issue, i am happy to take that task

Comment (rhysyngsun) [inline profiles/permissions_test.py]: Ok, fair enough
